<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1077</width>
    <height>659</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Сжатие машинного кода для RV32I набора команд</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab_3">
         <attribute name="title">
          <string>Информация о файле</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="1" column="2" colspan="2">
           <widget class="QTextEdit" name="unputFileInfoTextEdit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>300</width>
              <height>0</height>
             </size>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="maskFormatComboBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <item>
             <property name="text">
              <string>Словарь</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1 маска (32 бит)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2 маски (16/16 бит)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2 маски (24/8 бит)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>3 маски (16/8/8 бит)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>4 маски (8/8/8/8 бит)</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="2" colspan="2">
           <widget class="QLabel" name="label_6">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Информация о файле</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLabel" name="label_7">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Размер секции, Б:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QLineEdit" name="inputCodeSectionSizeLineEdit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Формат сжатой инструкции</string>
            </property>
            <property name="margin">
             <number>5</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QTextEdit" name="inputTextContentTextEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="tabStopWidth">
             <number>160</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="label_2">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Содержание секции &quot;.text&quot;</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Информация о сжатии</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_8">
                <item>
                 <widget class="QLabel" name="label_8">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Исходный размер секции кода (.text), Б</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="startTextSizeLineEdit">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_6">
                <item>
                 <widget class="QLabel" name="label_4">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Размер сжатой секции кода (.text), Б</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="cmprTextSizeLineEdit">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <item>
                 <widget class="QLabel" name="label_5">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Общий размер секций со словарем, Б</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="dictSizeLineEdit">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_3">
                <item>
                 <widget class="QLabel" name="label_9">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Размер сжатой секции кода + размер словарей , Б</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="totalSizeLineEdit">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_5">
                <item>
                 <widget class="QLabel" name="label_3">
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Коэффициент сжатия</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="cmprRateLineEdit">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="font">
                   <font>
                    <pointsize>14</pointsize>
                   </font>
                  </property>
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <spacer name="verticalSpacer_3">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Содержание словарей инструкций</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="1" column="2">
           <widget class="QTextEdit" name="dict1TextEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QTextEdit" name="dict2TextEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="4">
           <widget class="QTextEdit" name="dict3TextEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="5">
           <widget class="QTextEdit" name="dict4TextEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="dict1Name">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QLabel" name="dict2Name">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QLabel" name="dict3Name">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="5">
           <widget class="QLabel" name="dict4Name">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="chooseFileBtn">
          <property name="font">
           <font>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Выбрать файл</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="compressFileBtn">
          <property name="font">
           <font>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Сжать файл</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="decompressFileBtn">
          <property name="font">
           <font>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Расжать файл</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1077</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>О программе</string>
    </property>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
